### ISSUE TITLE
[explicit-module-boundary-types] does not recognise `export =`

### ISSUE THREAD

**Issue by SimenB** [typescript-eslint/typescript-eslint#1583]
**Repro**

```json
{
  "rules": {
    "@typescript-eslint/explicit-module-boundary-types": "error"
  }
}
```

```ts
class Class {
  bool() {
    return true;
  }
}

export = Class;
```

**Expected Result**

Warning about not typing the `bool` function.

**Actual Result**

No warning.

**Additional Info**

Doing `export default class Class {` will make it warn correctly.

**Versions**

| package                            | version   |
| ---------------------------------- | --------- |
| `@typescript-eslint/eslint-plugin` | `2.19.0`  |
| `@typescript-eslint/parser`        | `2.19.0`  |
| `TypeScript`                       | `3.7.5`   |
| `ESLint`                           | `6.8.0`   |
| `node`                             | `12.15.0` |
| `yarn`                             | `1.21.1`  |

**Comment by bradzacher** [comment]
This isn't just the rule recognising `export =` (it wasn't built to recognise it).

For your example, it also won't work because it doesn't track references right now.
The rule would similarly error on:
```ts
class Foo{}
export default Foo
```
